lib/evm_database.rb

```diff
@@ -112,4 +112,72 @@ def self.raise_server_event(event)
     msg = "Server IP: #{MiqServer.my_server.ipaddress}, Server Host Name: #{MiqServer.my_server.hostname}"
     MiqEvent.raise_evm_event_queue(MiqServer.my_server, event, :event_details => msg)
   end
+
+  def self.restart_failover_monitor_service
+    service = LinuxAdmin::Service.new("evm-failover-monitor")
+    service.restart if service.running?
+  end
+
+  def self.restart_failover_monitor_service_queue
+    MiqQueue.put(
+      :class_name  => name,
+      :method_name => 'restart_failover_monitor_service',
+      :role        => 'database_operations',
+      :zone        => nil
+    )
+  end
+
+  def self.run_failover_monitor(monitor = nil)
+    require 'manageiq-postgres_ha_admin'
+    ManageIQ::PostgresHaAdmin.logger = Vmdb.logger
+
+    monitor ||= ManageIQ::PostgresHaAdmin::FailoverMonitor.new(Rails.root.join("config", "ha_admin.yml").to_s)
+
+    configure_rails_handler(monitor)
+    configure_pglogical_handlers(monitor)
+
+    _log.info("Starting database failover monitor")
+    monitor.monitor_loop
+  end
+
+  def self.configure_rails_handler(monitor)
+    file_path = Rails.root.join("config", "database.yml").to_s
```

Comment:
I notice we're converting Pathnames to string here and on line 134.  Is it required?  Pathnames should just work.